pipeline-adapters/mlrun-pipelines-kfp-v1-8/src/mlrun_pipelines/patcher.py

```diff
@@ -68,22 +68,32 @@ def _create_enriched_mlrun_workflow(
                 )
                 return workflow
 
-        # enrich each pipeline step with your desire k8s attribute
         for kfp_step_template in workflow["spec"]["templates"]:
-            if kfp_step_template.get("container"):
-                for function_obj in functions:
-                    # we condition within each function since the comparison between the function and
-                    # the kfp pod may change depending on the attribute type.
-                    _set_function_attribute_on_kfp_pod(
-                        kfp_step_template,
-                        function_obj,
-                        "PriorityClassName",
-                        "priority_class_name",
-                    )
-                    _enrich_kfp_pod_security_context(
-                        kfp_step_template,
-                        function_obj,
-                    )
+            if "dag" in kfp_step_template:
```

Comment:
why not keeping the `kfp_step_template.get("container")` condition to make sure we have container settings instead of checking for dag?
This can be changed to
```
if not kfp_step_template.get("container"):
    continue
```
This verifies what we need without relying on kfp logic (that dag is the only case that there is no container)